### PR TITLE
Redo JSON documentation

### LINKED ISSUE #217: Document built-in System.Text.Json support
We currently only have docs for the Newtonsoft.Json plugin ([see page](https://www.npgsql.org/doc/types/jsonnet.html)).

### PR DESCRIPTION
**Pull request by roji**
@vonzshik @NinoFloris @brar would be good to get some eyes on this (but I'll go ahead and merge, we can always correct later).

Closes #217

/cc @ajcvickers

### CODE REVIEW

File: conceptual/Npgsql/types/json.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+```c#
+var dataSourceBuilder = new NpgsqlDataSourceBuilder(...);
+dataSourceBuilder.EnableDynamicJson();
+await using var dataSource = dataSourceBuilder.Build();
+```
+
+### [Global mapping](#tab/global)
+
+If you're using an older version of Npgsql which doesn't yet support `NpgsqlDataSource`, you can configure mappings globally for all connections in your application:
```
Comment (vonzshik) [inline conceptual/Npgsql/types/json.md:64]: Isn't that only required starting with 8.0, as 7.0 and 6.0 can just send them as is?

Comment (roji) [reply conceptual/Npgsql/types/json.md:64]: There's a note above: "Startin with Npgsql 8.0, to use this feature, you must fir eanble it by calling..." (typo with Startin...). Do you think that's not enough?

Comment (vonzshik) [reply conceptual/Npgsql/types/json.md:64]: I don't understand what's the point of this sentence:

```

Comment (roji) [reply conceptual/Npgsql/types/json.md:64]: Ah I see, OK - will remove it.

Comment (roji) [reply conceptual/Npgsql/types/json.md:64]: Or rather tweak it - there are people using Npgsql 8.0 but who are still not using NpgsqlDataSource

Comment (roji) [reply conceptual/Npgsql/types/json.md:64]: Done in #316

**Pull request merged**